docs/adr/002-handle-update.md

```diff
@@ -0,0 +1,99 @@
+# 1. What the UX of updates to state should be
+
+Date: 2023-01-27
+
+## Status
+
+Proposed
+
+## Context
+
+In general, Ansible modules allow two `states`, `present` and `absent`. When `state=present`, the resource is created if it does not already exist or is updated. In the case of updating a resource, the Ansible module provides `purge_*` parameters to give the user granular control over the information they wish to update.
+
+Let's take a deep look at how `purge_tags` work in the amazon.aws collection for simplicity:
+* When `purge_tags=true` and tags are set, existing tags are purged from the resource to match exactly what is defined by tags parameter.
+* If tags parameter is not set then tags are not modified even if `purge_tags=true`.
+* When `purge_tags=false` and tags are set, the resource tags are updated with what is defined by tags parameter.
+
+A number of existing Ansible modules support purge_* parameters for different configurations; just to mention a few examples:
+* [purge_listeners](https://github.com/ansible-collections/amazon.aws/blob/main/plugins/modules/elb_application_lb.py)
+* [purge_rules](https://github.com/ansible-collections/amazon.aws/blob/main/plugins/modules/elb_application_lb.py)
+* [purge_security_groups](https://github.com/ansible-collections/amazon.aws/blob/main/plugins/modules/rds_instance.py)
+* [purge_iam_roles](https://github.com/ansible-collections/amazon.aws/blob/main/plugins/modules/rds_instance.py)
+* [purge_policies](https://github.com/ansible-collections/amazon.aws/blob/main/plugins/modules/iam_user.py)
+* …
+
+For example, we have the following S3 bucket with some tags:
+
+```json
+    "s3_bucket_1": {
+        "Properties": {
+            "BucketName": "pravic-s3-bucket-1",
+            "Tags": [
+                {
+                    "Key": "tag_1",
+                    "Value": "Tag1"
+                },
+                {
+                    "Key": "tag_2",
+                    "Value": "Tag2"
+                }
+            ]
+        },
+        "Type": "AWS::S3::Bucket"
+    },
+```
+
+Suppose we want to update the S3 bucket, retaining the tags already assigned:
+
+```yaml
+    resources:
+        s3_bucket_1:
+            Type: AWS::S3::Bucket
+            Properties:
+            BucketName: pravic-s3-bucket-1
+            Tags:
+                - Key: tag_3
+                Value: Tag_3
+```
+
+Without `purge_tags` option, we would need to perform three tasks:
+* Gather information about the existing S3 bucket and find out its actual tags;
+* Remove the existing tags of the S3 bucket;
+* Re-tag the S3 bucket with the new and old tags;
+
+`purge_tags` option will allow you to set that configuration within only one task by setting `purge_tags=false`.
+
+## Decision and possible consequencies
```

Comment:
Most of this section feels like it really belongs in the Context section. There is a description of different approaches here, which is good, but we need to actually make a decision about what we're going to do in the ADR. I'd like to see a Decision section that clearly states how we're moving forward. If you want, you can just use a numbered list, like 1. add a purge_tags parameter and 2. the state parameter will accept the following values: present, absent, replaced. And while we don't need to go into details about the technical implementation, we need to provide enough information that someone else could come along and understand how it should work from the user's perspective.

For example, some questions that I expect this ADR should be able to answer are:
* if I have a resource that already exists, and I use `state=present` with a resource definition that would change the resource, what happens?
* if I have a resource definition for a resource that doesn't exist, and I use `state=replaced`, what happens?

Reply:
I didn't know we come up with a final decision, that's why I wasn't clearly stating what we have decided. I guess we want to comply to the network modules, right? The only difference is not using state=deleted, but use replaced with an empty list or None values.

I tweaked the doc in different sections, so let me know please if it's clearer now or if it requires additional changes.